Style category page

victornguyen opened the pull request:
Styling work done to CategoryPage components done as part of Hack Day.

Apologies in advance!

danni commented on src/pages/CategoryPage.js:
```diff
@@ -60,6 +61,8 @@ class CategoryPage extends React.Component {
                 <Router.RouteHandler />
                 <mui.AppBar title={this.category.name} />
 
+                <CategoryHeader />
```
In #67 I refactored HeaderBar to use for things like this. Because I think it's consistently the same styles.

danni commented on src/components/ResultTile.js:
```diff
@@ -53,11 +54,15 @@ class ResultTile extends React.Component {
-                <div>{object.name}</div>
-                <div>{object.site.name}</div>
+                <h2 className="name">{object.name}</h2>
+                <div className="site">{object.site.name}</div>
```
Can you merge this with master? The class here is site_name, which is being used in a test. Happy to have either, so long as the test is correct.

danni commented on src/components/ResultTile.js:
```diff
@@ -53,11 +54,15 @@ class ResultTile extends React.Component {
+                <ul className="service-list">{
+                    (object.service_types || []).map((service) => {
+                        return <li>{service}</li>;
+                    })
```
I believe you can do `map(service => <li>{service}</li>)`

Commit:
Google Maps API adaptor wraps the map components in Promises

Commit:
Wrap the geocoder in a promise

Commit:
Wrap browser geolocation in a Promise

Commit:
Update the UI in response to geolocation

Commit:
Get GOOGLE_KEY from environment

Add flow to HtmlDocument.

Commit:
Make locateMe a method

Commit:
Add state machine to take user through the entire geolocation process

 * Failure modes aren't tested yet

Commit:
Add icons

danni commented on src/components/CategoryHeader.js:
```diff
@@ -0,0 +1,21 @@
+                </a>
+            </div>
+        );
+    }
```
This probably doesn't need its own class IMO. Also you can get (some of) the live data for this field from `this.state.meta` on the category page.

danni replied:
Either include this in `CategoryPage` or use `HeaderBar` as a generic element.

DanielHeath replied:
this.state.meta is just a huge URL string

danni replied:
Oops. Someone should debug that.

On Friday, 21 August 2015, Daniel Heath notifications@github.com wrote:

> In src/components/CategoryHeader.js
> https://github.com/ask-izzy/ask-izzy/pull/68#discussion_r37605824:
> 
> > +
> > +export default class CategoryHeader extends React.Component {
> > +
> > -    render(): React.Element {
> > -        return (
> > -            <div className="CategoryHeader">
> > -                <div className="found">
> > -                    I found 25 housing services for men
> > -                    ages 25-54 in Brunswick
> > -                </div>
> > -                <a className="search-link" href="#">
> > -                    Try a different search
> > -                </a>
> > -            </div>
> > -        );
> > -    }
> 
> this.state.meta is just a huge URL string
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ask-izzy/ask-izzy/pull/68/files#r37605824.

danni commented on src/components/OpeningTimes.js:
```diff
@@ -35,21 +37,24 @@ class OpeningTimes extends React.Component {
-                    <span>Now open</span>&nbsp;
-                    <span>Closes {formatTime(todayOpen.close)}</span>
+                <div className="OpeningTimes">
+                    <icons.Clock className="coloured-icon" />
```
#71 includes a refactoring of the coloured icons to make it much easier to use.

DanielHeath committed:
Style the category list (originally by @victornguyen , cleaned up a little by me)

Commit:
Merge branch 'nav' into geolocation

Commit:
Redirect the user to the geolocation page if required

Commit:
Style the done button

Commit:
Enable changing the location; report error cleanly on page

Commit:
Build a location we can actually search with

Commit:
Fix a warning message about class/className

Commit:
Make enter submit the location form

It's possible we don't want this, because we will show auto-completion
options. But it's easy enough to take out.

Commit:
Add a progress icon to element saying you're being located

Commit:
Only show "find me" if geolocation is supported

Commit:
Make category page test pass again

Commit:
Enable flow in browser.js and webdriver.js

Commit:
Merge branch 'nav' into geolocation

Commit:
Fix nav test after merge

Commit:
Make it possible to only run tagged tests with ONLY and SKIP

Commit:
Write a test for successful geolocation

Commit:
Simulate a geolocation failure

Commit:
Fix Yadda when ONLY is not set

Commit:
Supply Google API credentials to Travis

Commit:
Write a test for the complete personalisation flow

Commit:
Pass script executed in the browser as a closure not a template string

Commit:
Reduce calls to await in the hope it reduces promise round trips

Commit:
Remove race in tests

danni committed:
Merge pull request #74 from ask-izzy/nav

Navigation between pages

danni committed:
Merge pull request #69 from ask-izzy/geolocation

Geolocation

Commit:
Merge remote-tracking branch 'origin/style-category-page' into result-list-styling

Commit:
Clean up icon usage
